server/src/main/java/org/opensearch/common/cache/tier/EhCacheDiskCachingTier.java

```diff
@@ -39,14 +46,22 @@
 import org.ehcache.event.EventType;
 import org.ehcache.expiry.ExpiryPolicy;
 import org.ehcache.impl.config.store.disk.OffHeapDiskStoreConfiguration;
-
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.common.unit.ByteSizeValue;
+import org.opensearch.search.query.QuerySearchResult;
+
+/**
+ * An ehcache-based disk tier implementation.
+ * @param <K> The key type of cache entries
+ * @param <V> The value type of cache entries
+ */
 public class EhCacheDiskCachingTier<K, V> implements DiskCachingTier<K, V> {
 
     // A Cache manager can create many caches.
-    private final PersistentCacheManager cacheManager;
+    private static PersistentCacheManager cacheManager = null;
```

Comment:
Why change to static and initialize to null?

Reply:
This was also for test case issues. When a test case like OpenSearchSingleNodeTestCase starts, it creates nodes which have IndicesRequestCache in them. Once we turned on the disk tier, that means it makes an Ehcache cache manager which locks its directory. Then when we try to make another disk tier to use in the test we can't instantiate it, since it tries to start its cache manager but the directory is locked. 

I don't think we can tell the original one to close, so instead we have one class-wide static manager per node, and when the tier is created it checks if there's already a manager and creates a new one if there's not. Maybe we could also avoid this problem by adding some unique UUID to the end of the storage path that the disk tier builder receives?